contracts/pools/PoolCollection.sol

```diff
@@ -1250,66 +1260,91 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
+
+                // ensure that the new target isn't higher than the previously established target
+                targetBNTTradingLiquidity = Math.min(targetBNTTradingLiquidity, newTargetBNTTradingLiquidity);
+            } else {
+                // the current BNT trading liquidity is not 0 - limit the target using the growth factor
+                targetBNTTradingLiquidity = Math.min(
+                    targetBNTTradingLiquidity,
+                    liquidity.bntTradingLiquidity * LIQUIDITY_GROWTH_FACTOR
+                );
             }
 
-            targetBNTTradingLiquidity = newTargetBNTTradingLiquidity;
-        } else if (targetBNTTradingLiquidity >= liquidity.bntTradingLiquidity) {
-            // if the target is above the current trading liquidity, limit it by factoring the current value up. Please
-            // note that if the target is below the current trading liquidity - it will be reduced to it immediately
-            targetBNTTradingLiquidity = Math.min(
-                targetBNTTradingLiquidity,
-                liquidity.bntTradingLiquidity * LIQUIDITY_GROWTH_FACTOR
-            );
+            bntTradingLiquidityDelta = targetBNTTradingLiquidity - liquidity.bntTradingLiquidity;
+            baseTokenTradingLiquidityDelta = MathEx.mulDivF(bntTradingLiquidityDelta, fundingRate.d, fundingRate.n);
+
+            return
+                TargetTradingLiquidity({
+                    update: true,
+                    bnt: targetBNTTradingLiquidity,
+                    baseToken: liquidity.baseTokenTradingLiquidity + baseTokenTradingLiquidityDelta
+                });
         }
 
-        // calculate the base token trading liquidity based on the delta between the new and the previous BNT trading
-        // liquidity and the effective funding rate (please note that the effective funding rate is always the rate
-        // between BNT and the base token)
-        uint256 bntTradingLiquidityDelta = targetBNTTradingLiquidity - liquidity.bntTradingLiquidity;
-        uint256 baseTokenTradingLiquidityDelta = bntTradingLiquidityDelta == 0
-            ? 0
-            : MathEx.mulDivF(bntTradingLiquidityDelta, fundingRate.d, fundingRate.n);
+        // liquidity decrease
+        // note that liquidity decrease isn't capped
+        bntTradingLiquidityDelta = liquidity.bntTradingLiquidity - targetBNTTradingLiquidity;
+        baseTokenTradingLiquidityDelta = MathEx.mulDivF(bntTradingLiquidityDelta, fundingRate.d, fundingRate.n);
 
         return
-            TradingLiquidityAction({
+            TargetTradingLiquidity({
                 update: true,
-                newBNTTradingLiquidity: targetBNTTradingLiquidity,
-                newBaseTokenTradingLiquidity: liquidity.baseTokenTradingLiquidity + baseTokenTradingLiquidityDelta
+                bnt: targetBNTTradingLiquidity,
+                baseToken: liquidity.baseTokenTradingLiquidity - baseTokenTradingLiquidityDelta
```

Comment:
I mas not convinced that the condition `liquidity.baseTokenTradingLiquidity >= baseTokenTradingLiquidityDelta` is mathematically guaranteed at this point (it might, but I cannot easily infer that from the code).
So as a safety measure, we might be better off with:
```
baseToken: MathEx.subMax0(liquidity.baseTokenTradingLiquidity, baseTokenTradingLiquidityDelta)
```
So if `liquidity.baseTokenTradingLiquidity < baseTokenTradingLiquidityDelta` under any circumstances, then the code above will not revert the transaction, and the resulting `baseToken: 0` will lead the calling function to reset the trading liquidity.

Reply:
Makes sense - updated.